core/server/src/main/java/alluxio/proxy/FileSystemClientRestServiceHandler.java

```diff
@@ -89,407 +98,375 @@ public FileSystemClientRestServiceHandler(@Context ServletContext context) {
-          options.setOwner(owner);
-        }
-        if (mode != null) {
-          options.setMode(mode);
-        }
-        if (recursive != null) {
-          options.setRecursive(recursive);
-        }
-        if (ttl != null) {
-          options.setTtl(ttl);
+        if (options == null) {
+          mFileSystem.setAttribute(new AlluxioURI(path), SetAttributeOptions.defaults());
+        } else {
+          mFileSystem.setAttribute(new AlluxioURI(path), options);
         }
-        if (ttlAction != null) {
-          options.setTtlAction(ttlAction);
-        }
-        mFileSystem.setAttribute(new AlluxioURI(path), options);
         return null;
       }
     });
   }
 
   /**
    * @summary unmount a path
-   * @param path the file path
+   * @param path the Alluxio path
+   * @param options method options
    * @return the response object
    */
   @POST
-  @Path(UNMOUNT)
+  @Path(PATHS_PREFIX + PATH_PARAM + UNMOUNT)
   @ReturnType("java.lang.Void")
-  public Response unmount(@QueryParam("path") final String path) {
+  public Response unmount(@PathParam("path") final String path,
+      final UnmountOptions options) {
     return RestUtils.call(new RestUtils.RestCallable<Void>() {
       @Override
       public Void call() throws Exception {
-        Preconditions.checkNotNull(path, "required 'path' parameter is missing");
-        mFileSystem.unmount(new AlluxioURI(path));
+        if (options == null) {
+          mFileSystem.unmount(new AlluxioURI(path), UnmountOptions.defaults());
+        } else {
+          mFileSystem.unmount(new AlluxioURI(path), options);
+        }
         return null;
       }
     });
   }
 
   /**
-   * @summary upload a file
-   * @param path the file path
-   * @param blockSizeBytes the target block size in bytes
-   * @param locationPolicy the location policy
-   * @param mode the octal mode to used for file permissions
-   * @param recursive whether parent directories should be created if they do not already exist
-   * @param ttl the time-to-live (in milliseconds)
-   * @param ttlAction action to take after TTL is expired
-   * @param writeType the write type to use
+   * @summary write a stream
+   * @param id the stream id
    * @param is the input stream
    * @return the response object
    */
   @POST
-  @Path(UPLOAD)
+  @Path(STREAMS_PREFIX + ID_PARAM + WRITE)
   @ReturnType("java.lang.Void")
   @Consumes(MediaType.APPLICATION_OCTET_STREAM)
-  public Response upload(@QueryParam("path") final String path,
-      @QueryParam("blockSizeBytes") final Long blockSizeBytes,
-      @QueryParam("locationPolicy") final String locationPolicy,
-      @QueryParam("mode") final Integer mode,
-      @QueryParam("recursive") final Boolean recursive,
-      @QueryParam("ttl") final Long ttl,
-      @QueryParam("ttlAction") final TtlAction ttlAction,
-      @QueryParam("writeType") final WriteType writeType,
+  public Response write(@PathParam("id") final Integer id,
       final InputStream is) {
```

Comment:
fixed everywhere (I use to keep one argument per line when we had a lot of them)